socket/client_test.go

```diff
@@ -56,3 +59,22 @@ func TestNewClientFail(t *testing.T) {
 		t.Fatalf("expected error, got nothing")
 	}
 }
+
+func TestDisconnectServer(t *testing.T) {
+	t.Parallel()
+	// start the server in server-client/server/main.go
+	client, err := NewClient(":8080")
```

Comment:
We run `go test` on every PR / push on main, so it's better to create a separate server inside this test function.

Also `:8080` is usually heavily used by other apps and I think _for testing_ some other random port would be more appropriate, since we don't want to have random failures because `8080` is taken. Maybe something like `12345`?